lib/compat.php

```diff
@@ -228,6 +228,27 @@ function gutenberg_get_post_type_capabilities( $user, $name, $request ) {
 	return $value;
 }
 
+/**
+ * Make sure oEmbed REST Requests apply the WP Embed security mechanism for WordPress embeds.
+ *
+ * TODO: This is a temporary solution. This code should be included in WordPress Core.
+ *
+ * @since ?
+ *
+ * @param  array $data The REST Request response.
+ * @return array       The REST Request response.
+ */
+function gutenberg_filter_oembed_result( $data ) {
+	if ( ! defined( 'REST_REQUEST' ) || false === REST_REQUEST || empty( $data['html'] ) || empty( $_GET['url'] ) ) {
+		return $data;
+	}
+
+	$data['html'] = wp_filter_oembed_result( $data['html'], (object) $data, $_GET['url'] );
+
+	return $data;
+}
+add_filter( 'oembed_response_data', 'gutenberg_filter_oembed_result', 14, 1 );
```

Comment:
The fourth argument here is redundant, since `1` is the default value if omitted.

https://developer.wordpress.org/reference/functions/add_filter/

Reply:
How did you arrive at the priority of `14`?

Reply:
I'll remove the fourth argument. About the priority: `wp_filter_oembed_result()` must happen after `get_oembed_response_data_rich()` which already hooks at 10 `oembed_response_data`. So 11 or upper works. 14 leaves some space for others to eventually filter in between.